src/Microsoft.ML.Tokenizers/Model/BertTokenizer.cs

```diff
@@ -0,0 +1,729 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.ML.Tokenizers
+{
+    /// <summary>
+    /// Tokenizer for Bert model.
+    /// </summary>
+    /// <remarks>
+    /// The BertTokenizer is a based on the WordPieceTokenizer and is used to tokenize text for Bert models.
+    /// The implementation of the BertTokenizer is based on the original Bert implementation in the Hugging Face Transformers library.
+    /// https://huggingface.co/transformers/v3.0.2/model_doc/bert.html?highlight=berttokenizerfast#berttokenizer
+    /// </remarks>
+    public sealed partial class BertTokenizer : WordPieceTokenizer
+    {
+        internal BertTokenizer(
+                    Dictionary<StringSpanOrdinalKey, int> vocab,
+                    Dictionary<int, string> vocabReverse,
+                    PreTokenizer? preTokenizer,
+                    Normalizer? normalizer,
+                    IReadOnlyDictionary<string, int>? specialTokens,
+                    bool doLowerCase,
+                    bool doBasicTokenization,
+                    bool splitOnSpecialTokens,
+                    string unknownToken,
+                    string sepToken,
+                    string padToken,
+                    string clsToken,
+                    string maskToken,
+                    bool tokenizeChineseChars,
+                    bool stripAccents) : base(vocab, vocabReverse, preTokenizer, normalizer, specialTokens, unknownToken)
+        {
+            DoLowerCase = doLowerCase;
+            DoBasicTokenization = doBasicTokenization;
+            SplitOnSpecialTokens = splitOnSpecialTokens;
+
+            SepToken = sepToken;
+            SepTokenId = vocab[new StringSpanOrdinalKey(sepToken)];
+
+            PadToken = padToken;
+            PadTokenId = vocab[new StringSpanOrdinalKey(padToken)];
+
+            ClsToken = clsToken;
+            ClsTokenId = vocab[new StringSpanOrdinalKey(clsToken)];
+
+            MaskToken = maskToken;
+            MaskTokenId = vocab[new StringSpanOrdinalKey(maskToken)];
+
+            TokenizeChineseChars = tokenizeChineseChars;
+            StripAccents = stripAccents;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the tokenizer should lowercase the input text.
+        /// </summary>
+        public bool DoLowerCase { get; }
```

Comment:
Have we taken any of these tokenizers through API review? I suspect we'd prefer different names for some of these properties, for example.

Reply:
I'll try to find out if I can get spot soon and review it.